packages/runtime/container-runtime/src/opLifecycle/outbox.ts

```diff
@@ -304,7 +303,10 @@ export class Outbox {
 		}
 
 		const rawBatch = batchManager.popBatch();
-		if (rawBatch.hasReentrantOps === true && this.params.config.enableGroupedBatching) {
+		if (
+			rawBatch.hasReentrantOps === true &&
+			this.params.groupingManager.shouldGroup(rawBatch)
```

Comment:
The goal here is to minimize risk and rebasing is required for grouped batching. So, if the batch is reentrant but it will not get grouped, there's no point in rebasing it..

Reply:
Is it the case? Do we stamp individual ops in a batch with different reference Sequence Numbers? I'm looking at Container.submitBatch(), and while IBatchMessage used there has referenceSequenceNumber, it's ignored and a single referenceSequenceNumber is stamped on whole batch.
So I think rebasing is required in all cases, unless we are changing that flow.
That said, we have been using wrong referenceSequenceNumbers all along in such cases. So point could be made that we keep limping along. I do not know how to assess risk here - it's quite possible that even though we do see issues with rebasing logic, it could be that we see them only because we are looking! And the old flow of submitting batches with wrong reference sequence numbers were also causing troubles but we simply either have no instrumentation to see it (issues with eventual consistency) OR we can't isolate these sessions and signal simply is to weak to notice it in total noise.

Reply:
@vladsud, as of right now, rebasing is not engaged at all if op grouping is not enabled. This change does not regress any behavior at all. What it does is to maintain the _current_ logic of not rebasing unless op grouping happens.